numpy/lib/index_tricks.py

```diff
@@ -596,6 +597,419 @@ def __iter__(self):
     next = __next__
 
 
+class ndrange(_nx.collections_abc.Sequence):
+    """
+    The ndrange type represents an immutable sequence of numbers and can be
+    used for looping over specific multi-dimensional indicies in ``for`` loops.
+
+    # TODO: how do I get this to show up in the docs
+    class ndrange(stop)
+    class ndrange(start, stop[, step])
+
+    The arguments to the ``ndrange`` constructor must be iterables (lists,
+    tuple, 1D-arrays, etc.) of integers. Individual integers are also accepted
+    and are coerced to tuples of length 1.
```

Comment:
I think this is unnecessary to say, since you also specify the argument types in the parameters section.

Reply:
I guess I mostly just reworded the Python docs for this. 

I kinda agree that numpy docs typically read better, but I don't know if we want that for this particular function.

I can rewrite these docs in the style of numpy if you wish.

Reply:
I think that since we are essentially extending python's range, it is fine to leave some of their details out since users can always get the idea from the range docstring. Plus numpy has the parameters section.